lib/Entity/Schedule.php

```diff
@@ -841,11 +865,23 @@ public function save($options = [])
         }
 
         // Update schedule criteria
+        $criteriaIds = [];
         foreach ($this->criteria as $criteria) {
             $criteria->eventId = $this->eventId;
             $criteria->save();
+
+            $criteriaIds[] = $criteria->id;
         }
 
+        // Remove records that no longer exist.
+        $this->getStore()->update('
+            DELETE FROM `schedule_criteria` 
+             WHERE `id` NOT IN (' . implode(',', $criteriaIds) . ')
```

Comment:
This probably wants to be executed if we do have any `$criteriaIds` to not upset sql